orbit/pkg/update/notifications.go

```diff
@@ -499,6 +509,33 @@ func (w *windowsMDMBitlockerConfigFetcher) attemptBitlockerEncryption(notifs fle
 		return
 	}
 
+	// if the disk is encrypted, try to decrypt it first.
+	if encryptionStatus != nil &&
+		encryptionStatus.ConversionStatus == bitlocker.ConversionStatusFullyEncrypted {
+		log.Debug().Msg("disk was previously encrypted. Attempting to decrypt it")
+
+		if err := w.decryptVolume(targetVolume); err != nil {
+			log.Error().Err(err).Msg("decryption failed")
+
+			// TODO: discuss: should we continue and try to encrypt the disk anyways instead?
+			if serverErr := w.updateFleetServer("", err); serverErr != nil {
+				log.Error().Err(serverErr).Msg("failed to send decryption failure to Fleet Server")
+				return
+			}
+		}
+
+		// return regardless of the operation output.
+		//
+		// - if we failed to decrypt the disk, we don't want to continue
+		// - if we started the decryption process, it might take a while for the
+		//   disk to be fully decrypted
+		//
+		// TODO:
+		// - revisit this logic in code review with the team
+		// - should we do `w.lastRun = time.Now()` to retry again in an hour?
```

Comment:
calling out this TODO

Reply:
It seems like it shouldn't be a problem to retry this logic on the next tick, given that if it is still decrypting, it would be a no-op (due to `bitLockerActionInProgress`)? So in that case I think it's beneficial to _not_ set `w.lastRun`, so that as soon-ish as the decryption is over with, it can go ahead and encrypt.

Reply:
agreed, I added a note basically copying what you're saying here for our future selves. Thank you!